src/components/common/PinButton.tsx

```diff
@@ -39,7 +39,7 @@ const PinButton = ({ item, type, onClick }: Props): JSX.Element => {
 
   return (
     <GraaspPinButton
-      type={type}
+      type={type as ActionButtonVariant}
```

Comment:
Is this caused by the nullable? If yes, consider using a non-nullable prop type for `type`, or returning `null` if not set (but I think removing the nullable makes sense).